AsyncDisplayKit/ASCollectionView.mm

```diff
@@ -605,7 +635,7 @@ - (ASSizeRange)dataController:(ASDataController *)dataController constrainedSize
   }
   
   if (_asyncDelegateImplementsInsetSection) {
-    sectionInset = [_asyncDelegate collectionView:self layout:self.collectionViewLayout insetForSectionAtIndex:indexPath.section];
+    sectionInset = [(id<ASCollectionViewDelegateFlowLayout>)_asyncDelegate collectionView:self layout:self.collectionViewLayout insetForSectionAtIndex:indexPath.section];
```

Comment:
I want to move our reliance on cell size modification (like sectionInsets) to that of the layout. UICollectionView only exposes this delegate method on `UICollectionViewDelegateFlowLayout`, which essentially makes this functionality valuable only when working with flow layouts.

I could see the issue where a custom layout author provides a custom delegate that implements `collectionView:layout:insetForSectionAtIndex`, hoping that ASDK will see the implementation and use it for the internal section inset calculation. However, this could easily be supported by composing `ASCollectionViewDelegateFlowLayout` into the custom delegate protocol. It's a bit of an edge case, but if you're interested in supporting this pathway we could create a simple protocol with just this method definition that would be composed into `ASCollectionViewDelegateFlowLayout`.